File: src/main/java/net/flyingfishflash/loremlist/api/MaintenanceApiServiceDefault.java
package net.flyingfishflash.loremlist.api;

import net.flyingfishflash.loremlist.api.data.response.ApiServiceResponse;
import net.flyingfishflash.loremlist.api.data.response.DomainPurgedResponse;
import net.flyingfishflash.loremlist.domain.ServiceResponse;
import net.flyingfishflash.loremlist.domain.maintenance.MaintenanceService;
import net.flyingfishflash.loremlist.domain.maintenance.data.DomainPurged;
import org.springframework.stereotype.Service;

@Service
public class MaintenanceApiServiceDefault implements MaintenanceApiService {

  private final MaintenanceService maintenanceService;

  public MaintenanceApiServiceDefault(MaintenanceService maintenanceService) {
    this.maintenanceService = maintenanceService;
  }

  @Override
  public ApiServiceResponse<DomainPurgedResponse> purge() {
    ServiceResponse<DomainPurged> serviceResponse = maintenanceService.purge();
    DomainPurged domainPurged = serviceResponse.getContent();
    DomainPurgedResponse domainPurgedResponse =
        new DomainPurgedResponse(
            domainPurged.associationDeletedCount(),
            domainPurged.itemDeletedCount(),
            domainPurged.listDeletedCount());
    return new ApiServiceResponse<>(domainPurgedResponse, serviceResponse.getMessage());
  }
}

File: src/main/kotlin/net/flyingfishflash/loremlist/api/MaintenanceApiServiceDefault.kt
package net.flyingfishflash.loremlist.api

import net.flyingfishflash.loremlist.api.data.response.ApiServiceResponse
import net.flyingfishflash.loremlist.api.data.response.DomainPurgedResponse
import net.flyingfishflash.loremlist.domain.maintenance.MaintenanceService
import org.springframework.stereotype.Service

@Service
class MaintenanceApiServiceDefault(private val maintenanceService: MaintenanceService) : MaintenanceApiService {
  override fun purge(): ApiServiceResponse<DomainPurgedResponse> {
    val serviceResponse = maintenanceService.purge()
    val domainPurgedResponse =
      DomainPurgedResponse(
        associationDeletedCount = serviceResponse.content.associationDeletedCount,
        itemDeletedCount = serviceResponse.content.itemDeletedCount,
        listDeletedCount = serviceResponse.content.listDeletedCount,
      )
    return ApiServiceResponse(content = domainPurgedResponse, message = serviceResponse.message)
  }
}
